.github/workflows/php-code-coverage.yml

```diff
@@ -0,0 +1,76 @@
+name: PHP code coverage
+
+on:
+  pull_request
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+    name: PHP=8.1, WP=6.6, WC=9.1.2
+    env:
+      PHP_VERSION: '8.1'
+      WP_VERSION: '6.6'
+      WC_VERSION: '9.1.2'
+    steps:
+      - name: Get Changed Files
+        id: get-changed-files
+        uses: hanseltimeindustries/get-changed-files@v1
+        with:
+          format: 'space-delimited'
+          filter: '**/includes/**/*'
+      
+      - name: Checking out
+        if: ${{ steps.get-changed-files.outputs.all }} # Only run if there are changed files
+        uses: actions/checkout@v2
+
+      - name: Set up dependencies caching
+        if: ${{ steps.get-changed-files.outputs.all }}
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/composer/
+          key:  ${{ runner.os }}-composer-${{ hashFiles('composer.lock') }}
+
+      - name: Set up PHP
+        if: ${{ steps.get-changed-files.outputs.all }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.1'
+          coverage:    xdebug
+
+      - name: Set up PHPUnit 9.5 for compatibility
+        if: ${{ steps.get-changed-files.outputs.all }}
+        run: wget https://phar.phpunit.de/phpunit-9.5.28.phar && mv phpunit-9.5.28.phar phpunit.phar
+        
+      - name: Run CI checks
+        if: ${{ steps.get-changed-files.outputs.all }}
+        run: bash bin/run-ci-tests.bash
+        
+      - name: Code Coverage Report
+        if: ${{ steps.get-changed-files.outputs.all }}
+        uses: irongut/CodeCoverageSummary@v1.3.0
+        with:
+          filename: cobertura.xml
+          format: markdown
+          hide_branch_rate: false
+          hide_complexity: true
+          indicators: true
+          output: both
+          thresholds: '60 80'
+          
+      - name: Filter Changed Files in Report
+        if: ${{ steps.get-changed-files.outputs.all }}
+        run: |
+          echo "### 📈 PHP Unit Code Coverage Report" > code-coverage-results-filtered.md 
```

Comment:
I also tried to put this into a single statement, but for some unknown reason, some parts got cut, so I had to revert to multiple lines.